### DISCUSSION TITLE
Updating Scarpet docs

### DISCUSSION THREAD

**Discussion by Ghoulboy78** [gnembon/fabric-carpet#837] [Ideas]
This is a relatively monumental task, so would have to be split within the separate files, each being updated with relevant docs and examples etc:

### API docs
- [x] Auxiliary.md (#1415)
- [x] BlockIterations.md (#1754)
- [x] BlockAndWorldAccess.md (#1757)
- [ ] Entities.md
- [ ] Events.md
- [ ] Inventories.md
- [ ] Overview.md
- [ ] Scoreboard.md
- [ ] ScriptCommand.md

### Language docs

- [ ] Containers.md
- [ ] FunctionsAndControlFlow.md
- [ ] LoopsAndHigherOrderFunctions.md
- [ ] Math.md
- [ ] Operators.md
- [ ] Overview.md
- [ ] SystemFunctions.md
- [ ] VariablesAndConstants.md

The idea is that anyone can reply here with one of the above files that they want to update

### Rules for updating

Each function should have  a detailed description with at least one example (the more the merrier obviously)

Each title description should have a detailed description of the section name.

**Comment by altrisi** [comment]
IMO if the whole docs are getting upgraded, we should also make a better doc format/framework than a simple markdown file, though that would be an even larger project.

**Comment by altrisi** [comment]
Also another thing, what do you think about changing code snippets to use code blocks with (py? js?) syntax highlighting instead of `pre`s?

**Reply by Ghoulboy78** [reply]
Well, the reason I use `<pre>` is because I don't have grave accent on my keyboard, so it's harder. Also all the other examples use `<pre>` and they don't use any specific language

**Reply by James103** [reply]
At least on my end, if code is surrounded in `<pre> </pre>`, it appears in a multiline format, but if the code is surrounded in `<code> </code>` tags, anything in-between appears as inline code.

Example input:
```
<code> Inline code </code>

<pre> Multiline code </pre>
```

Output from above example:
<code> Inline code </code>

<pre> Multiline code </pre>

**Reply by altrisi** [reply]
Ik right now it's all `pre`, that's why I'm suggesting the change since this disc is about updating docs.

IMO this this just looks way better:

```js
run_program() -> (
  loop( 10,
    // looping 10 times
    // comments are allowed in scripts located in world files
    // since we can tell where that line ends
    foo = floor(rand(10));
    check_not_zero(foo);
    print(_+' - foo: '+foo);
    print('  reciprocal: '+  _/foo )
  )
);
check_not_zero(foo) -> (
  if (foo==0, foo = 1)
)
```

than this:

<pre>
run_program() -> (
  loop( 10,
    // looping 10 times
    // comments are allowed in scripts located in world files
    // since we can tell where that line ends
    foo = floor(rand(10));
    check_not_zero(foo);
    print(_+' - foo: '+foo);
    print('  reciprocal: '+  _/foo )
  )
);
check_not_zero(foo) -> (
  if (foo==0, foo = 1)
)
</pre>

Not necessarily has to be done at the same time, just to keep the suggestion so it's not lost.

**Reply by altrisi** [reply]
Re James
With code blocks I meant \`\`\`, sorry, didn't even know `<code>` was a thing.

**Reply by Ghoulboy78** [reply]
I see your point here
